src/Sql/Vault/dbo/Stored Procedures/Cipher/Cipher_Move_V2.sql

```diff
@@ -0,0 +1,36 @@
+﻿CREATE PROCEDURE [dbo].[Cipher_Move_V2]
+    @Ids AS [dbo].[GuidIdArray] READONLY,
+    @FolderId AS UNIQUEIDENTIFIER,
+    @UserId AS UNIQUEIDENTIFIER
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    DECLARE @UserIdKey VARCHAR(50) = CONCAT('"', @UserId, '"')
+    DECLARE @UserIdPath VARCHAR(50) = CONCAT('$.', @UserIdKey)
+
```

Comment:
Might be more efficient to join here also:

;WITH [IdsToMoveCTE] AS (
    SELECT
        UCDV2.[Id]
    FROM
        [dbo].[UserCipherDetails_V2](@UserId) UCDV2
    INNER JOIN
        @Ids I
    ON
        UCDV2.[Id] = I.Id
)
UPDATE C
SET
    [Folders] =
        CASE
        WHEN @FolderId IS NOT NULL AND [Folders] IS NULL THEN
            CONCAT('{', @UserIdKey, ':"', @FolderId, '"', '}')
        WHEN @FolderId IS NOT NULL THEN
            JSON_MODIFY([Folders], @UserIdPath, CAST(@FolderId AS VARCHAR(50)))
        ELSE
            JSON_MODIFY([Folders], @UserIdPath, NULL)
        END
FROM
    [dbo].[Cipher] C
INNER JOIN
    [IdsToMoveCTE] CTE
ON
    C.[Id] = CTE.[Id]